---
title: "BLAST practical"
author: "Philip Ashton"
date: '`r format(Sys.time(), "%d %B, %Y")`'
---

<style>
body {
text-align: justify}
</style>

```{r settings, include = FALSE}
switch <- FALSE
```

# Learning objectives {-}
To become familiar with:  

1. The mechanics of running BLAST searches  

2. How to interpret BLAST results 

3. Ways in which homology can complicate the interpretation of BLAST results  

4. How BLAST, combined with the use of the extra information in graph genomes can aid in the localisation of AMR determinants to genomic elements

# Exercise 1 {-}

In this exercise you will be given 6 assembled Salmonella genomes. You need to identify which serotype they are based on the results of BLAST analysis against a custom database, and cross checking the results with (a shortened version of) the Kauffman-White scheme.

1. In the terminal, navigate to the `OTN_bioinfx/Day3/AM/Salmonella_serotyping` directory, there should be 6 genome assemblies (fasta format) labelled `Salm_1` to `Salm_6` and a `salmonella_antigens.fasta` file. <br>
In order to run BLAST, we need a query and a database. In this case, the database is going to be the `salmonella_antigens.fasta` file, and the queries are going to be the 6 Salmonella assemblies. For simplicity, we will process them one at a time.

2.  First we need to process the FASTA which will be our database so that it is compatible with BLAST. For this we use the makeblastdb program. <br>

```{bash, eval = FALSE}
makeblastdb -dbtype nucl -in salmonella_antigens.fasta
```
  
**Q1. How many sequences are there in the `salmonella_antigens.fasta` file? What are their names?**  

<div class="toggle"><button>Hint</button>
Since we know that the each fasta record has a header, and each header starts with a '>'. We can use `grep ‘>’ salmonella_antigens.fasta` to show all the sequence headers in the file. **CAREFUL** - make sure to include the quotation marks around the > in that command, or you will overwrite the contents of `salmonella_antigens.fasta`.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

3. Then, we can BLAST our query against the database we just made. Note that we use `blastn` here as we are comparing nucleotides.

3.1. Run

```{bash, eval = FALSE} 
blastn -help
```

**Q2. Look at the `*** Formatting options` section of the `blastn -help` output. We are going to load our data into Excel; which do you think is the most suitable `-outfmt` option to use?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

3.2. Run

```{bash, eval = FALSE} 
blastn -query Salm_1.fasta -db salmonella_antigens.fasta -outfmt XXX -out Salm_1_vs_salmonella_antigens.tsv
``` 

where XXX is the outfmt option you chose in question 3.1

3.3. Then, examine the output file using `head`

```{bash, eval = FALSE} 
head Salm_1_vs_salmonella_antigens.tsv
```

**Q3. How many columns does the output have?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q4. How many lines of the file has `head` shown you? How do you only show the top 4 lines of the file?**

<div class="toggle"><button>Hint</button>
[Google it!](http://www.letmegooglethat.com/?q=how+do+i+increase+the+number+of+lines+shown+by+head+linux)
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>


**Q5. What is separating the columns?**
<div class="toggle"><button>Hint</button>
What does the T in tsv stand for?
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

3.4. Run the blastn command without including the `-outfmt` option

**Q6. What happens if you run the command without the `-outfmt` option, or with different `-outfmt` options?**

<div class="toggle"><button>Hint</button>
Run the command with different outfmt options and look at the output file using `head` or `cat`.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

3.5. Run blastn for each genome against the 

3.6.  Make sure that each of your output files (e.g. `Salm_1_vs_salmonella_antigens.tsv`) is in the best format for viewing in Excel before proceeding.

**Q7. What kind of `blast` would we do if we were comparing proteins against `Salm_1.fasta`? What kind of BLAST would we do if we were comparing `salmonella_antigens.fasta` against `Salm_1.fasta`?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

4. Now, we are going to look at the BLAST output using Excel. For this, the output files (e.g. `Salm_1_vs_salmonella_antigens.tsv`) need to be on your local computer rather than your CLIMB instance.

4.1. Use Filezilla to move the output files onto your local computer and open them one at a time using Excel.

4.2. You might notice that `blastn` does not include column names in the output, these are the column headings, paste them into row 1 in Excel.

```{r table2, echo=FALSE, message=FALSE, warnings=FALSE, results='asis'}
tabl <- "

| Columns         | 
|-----------------|
| query_id        | 
| subject_id      |
| pct_identity    |
| aln_length      | 
| n_of_mismatches |
| gap_openings    |
| q_start         |
| q_end           |
| s_start         | 
| s_end           |
| e_value         |
| bit_score       |"

cat(tabl) # output the table in a format good for HTML/PDF/docx conversion
```
<div class="toggle"><button>Hint</button>
If you are having trouble with changing the orientation of the headers, use Paste Special -> transpose in Excel.
</div><br>

**Q8 What do the different column headings mean?**

<div class="toggle"><button>Hint</button>
Use Google to find out. For example, [here](http://www.metagenomics.wiki/tools/blast/blastn-output-format-6) or [here](https://www.ncbi.nlm.nih.gov/books/NBK279690/)
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q9. How many BLAST hits are there for `Salm_1` vs `salmonella_antigens?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q10. Look at the help for `blastn`, can you figure out some additional command line parameters you could enter which would reduce the number of results you have to look at? What are the potential downsides of applying these additional parameters?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q11. Which is the best column to sort by so that you see the best hits first? Sort the data by this column.**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q12. There are multiple different alleles of `fljB` and `fliC` in the BLAST database. Look at your BLAST results to identify the best fljB and fliC hits for each genome, and then lookup those alleles in the Salmonella Serotypes excel file to identify the serotype of each genome. Do this for all 6 genomes.** 

<div class="alert alert-info">
Reminder: fliC is H1 and fljB is H2.
</div>

<div class="toggle"><button>Hint</button>
One of the genomes might not have both H1 and H2.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

6. Take the BLAST results for one of the genomes, e.g. Salm_6. Filter the BLAST results so that only hits where the subject was either the ‘best’ `fljB` or ‘best’ `fliC`. You should have `6` blast hits. 

**Q13. Why are there 6 results? Hint1: Look at the query start and query end values for the `fljB` and `fliC` hits.**

<div class="toggle"><button>Hint</button>
There is an area of sequence similarity between `fljB` and `fliC`.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q14. What is another way you could tell which serotype the genomes are? The above method, looking at the genes which determine the serotype could be described as the ‘biologist’ way of serotyping from the genome, what might the ‘computer scientist/bioinformatician’ way be?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

# Exercise 2 {-}

In exercise 2, you will use BLAST within the Bandage genome graph visualiser to determine whether antibiotic resistance determinants are carried on plasmids or chromosomes. 

1. You will start by carrying out some basic data exploration on the command line of your remote instance. Locate the files in `OTN_bioinfx/Day3/AM/Salmonella_graph_genome_exercise`.

**Q1. What are the longest and shortest sequences in `genes_and_plasmids.fasta`?**

<div class="toggle"><button>Hint</button>
Use google to tell you how to do this. [this answer](https://www.biostars.org/p/118954/#119009) works well 
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q2. What are the differences between the sequences which have a name beginning with ‘p’ (i.e. lowercase p) and those which don’t?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q3. Either using Google or your microbiology knowledge, which genes in the file are typically plasmid-borne and which are typically chromosomal?**

<div class="toggle"><button>Hint</button>
Use `grep ‘>’ genes_and_plasmids.fasta` or the output of bioawk from above, to find the names of the sequences in the file.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

2. Using filezilla, download the 3 FASTG files and genes_and_plasmids.fasta from `Day3/AM/Salmonella_graph_genome_exercise` to your local computer

3. Load the first sample into Bandage following the instructions from the [Bandage github page](https://github.com/rrwick/Bandage/wiki/Getting-started#load-a-graph)

**Q4. What are the characteristics of this assembly?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q5. Do you think the assembly used an appropriate range of k-mer sizes? What other datasets do we need to get a better answer to this question?**

<div class="toggle"><button>Hint</button>
Look at [this](https://github.com/rrwick/Bandage/wiki/Effect-of-kmer-size)
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q6 Can you see any sequences which look like they could be plasmids?**

6. Prepare the FASTG file for BLAST analysis by following the instructions from [here](https://github.com/rrwick/Bandage/wiki/BLAST-searches)

7. BLAST the `genes_and_plasmids.fasta` against the genome loaded into Bandage. You can reduce the view to include only blast hits by changing the 'scope' drop down in the 'graph drawing' section, select 'around blast hits', change the distance to e.g. 2 or 3, and click draw graph. The distance controls how many contigs will be included in the view. Setting distance to 3 means that all contigs which are within 2 or 3 'jumps' of a contig with a blast hit will be included. Experiment with this parameter to see what effect having it at e.g. 0 and 10 has.

8. Inspect the genome graph and the BLAST hits, explore the different options Bandage has for ‘scope’, zoom in and out, Node labels, Font etc.
    
**Q7 For each genome, which plasmids and AMR genes are present in the sample?**

<div class="toggle"><button>Hint</button>
It might be easier to view the BLAST results via the ‘Create/view BLAST search results’
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Q8 If a plasmid is present, what does the ‘depth’ node label (combined with the graph structure) tell you about repeat sections in the plasmid?**

**Q9. In the assembly SRR5451253, what is interesting about the CTX-M15 encoding gene?**

<div class="toggle"><button>Hint</button>
blast every contig linked to the contig which encodes ctx-M15, you can do this by copying the node sequence or by using BLAST directly through Bandage via the Output tab. 
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

**Bonus question Q10: why do you think OmpD is included in the genes_and_plasmids.fasta?**

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

# Bonus Exercise {-}

If you have finsihed all the above exercises in good time, then feel free to have a go at this exercise, carrying out a Needleman-Wunsch alignment by hand. You can use paper or Excel, whichever you prefer.

1. Establish a scoring system - e.g. match = 1, mismatch = -1, gap = -1
2. You will be aligning two sequences `GAATTAA` and `GATTACA`. Create a scoring matrix like the below:   
![Scoring matrix](https://imgur.com/7JAQ6c9.png)
3. Fill in the table according to the instructions [here](https://en.wikipedia.org/wiki/Needleman%E2%80%93Wunsch_algorithm#Filling_in_the_table)
4. Trace back the arrows from the bottom right cell to the top left following the rules [here](https://en.wikipedia.org/wiki/Needleman%E2%80%93Wunsch_algorithm#Tracing_arrows_back_to_origin)
5. Check your answer by putting the sequences into [this](http://experiments.mostafa.io/public/needleman-wunsch/) website

 
<script>
  $(".toggle").click(function() {
    $(this).toggleClass("open");
  });
</script>