---
title: "Outbreak practical"
author: "Philip Ashton"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
        html_document:
                css: style.css
                theme: cerulean
                toc: yes
                toc_depth: 1
                number_sections: yes
                toc_float: yes
                fig_width: 7
                fig_height: 6
                fig_caption: yes
                self_contained: yes
                smart: no
        pdf_document:
                toc: yes
                number_sections: no
---
<style>
body {
text-align: justify}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Learning objectives {-}

1. Be able to interpret SNP distance histograms.

2. Be able to build phylogenetic trees from snippy output.

3. Be able to interpret phylogenetic trees for epidemiology.

4. Be familiar with various tree visualisation options.

5. Have a grasp on the relationship between trees and SNPs.

# Part 1 {-}

Inject 1 - In the aftermath of a massive earthquake in Haiti, the poorest country in the Americas, there has been a dramatic and worrying increase in the number of cases of cholera, the disease caused by the bacterium *Vibrio cholera*. This is the first time in more than 100 years than there has been a cholera outbreak in Haiti. Urgent efforts are underway to identify the source of the outbreak. Fortunately, in addition to field epidemiologists carrying out shoe-leather investigations in the affected areas, we have WGS of some of the isolates from the outbreak. You are the lead bioinformatician in charge of analysing this data. You need to process and interpret the data, and feedback to the public health doctors in charge of outbreak response to provide them with the fullest picture possible.

From using Google, or your microbiology knowledge about cholera:

* Why might there be an increase in cholera cases following an event like an earthquake?

* What is unusual about the genome of *V. cholera*?

On your CLIMB instance,

* Navigate to `~/data/outbreak_practical/phylo/dataset1`. 

* Run `ls` in this directory. This directory contains the alignment of all the high quality variable sites found in the core genome a for 19 *V. cholera* genomes from Haiti and 20 other *V. cholera* genomes from around the world by `snippy-core`.

* Calculate the pairwise SNP distance between all the isolates and plot the distribution as a histogram: 

```{bash eval=FALSE}
disty dataset1.aln | python ~/scripts/process_disty_output.py | histogram.py -b 30 --percentage
```

**Question 1:** How many processes are happening in this command? 

<div class="toggle"><button>Hint</button>
The `|` character is the unix command for ‘piping’ data between different processes.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - 3

**Question 2:** What does each stage of the process do? 

<div class="toggle"><button>Hint</button>
Run the different processes of the command, building up to the full command.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - 
Information on `disty` is here and histogram.py is here. `process_disty_output.py` is a script I wrote for data format conversion.

**Bonus question:** If you know any python, can you figure out what each line of `process_disty_output.py` is doing?


**Question 3:** What is the median pairwise SNP distance? How well do you think this describes the dataset?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - Median pairwise SNP distance is 61 SNPs. It doesn’t desribe the distribution well because there are three peaks in the distribution.

**Question 4: ** What does the histogram tell you about the relatedness of the samples in your dataset?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - there are three peaks in the distribution, one at low SNP distance (0-10), one at intermediate (50-60) and one at high(ish) (140-151). The low SNP distance pairs represent very closely related isolates, like we might see in an outbreak, the medium and high pairs represent the distance between different clades in the tree, and the population structure within the tree. See picture in outbreak practical directory.

**Question 5:**  Why are there 780 ‘samples’ in the output?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - there are 40 samples in the comparison. In the upper-triangular  pairwise distance matrix, not including self-self comparisons there are `N * (N - 1) / 2` comparisons; For `N = 40` this is `40 * 39 / 2 = 780.`

* Run IQ-TREE to generate a Maximum Likelihood tree. When IQ-TREE is run like this, it will also test a wide range of nucleotide substitution models to see which best fits the data. For larger alignments this can be a very time consuming step, so if speed is important, it might make sense to omit this step and use a general model (e.g. GTR).
`iqtree -s dataset1.aln -nt AUTO`

**Question 6:** Read the IQTREE output, which model was selected as the best? What is the full name of the model? I.e. what do the initials stand for?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer -  “K3P+ASC chosen according to BIC” Kimura 3 Parameter
** Question 7:** How many threads were selected as best? Why might the maximum number of threads not be selected?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - 1 (on my computer anyway). Because I/O takes time, so not always quickest to divide tasks between different processors.

* Download the tree (the file you want is the one which ends in `.treefile`) to your local computer, you can either do this with filezilla or by using `cat` to print the contents of the file to screen and then copy and pasting it into a new file on your local computer. **Change the file extension of the tree file to `.nwk`.**

* Navigate to https://microreact.org/upload and upload the `dataset1.aln.nwk` file and the `geog_loc_microreact.csv` file. The microreact interface is hopefully quite intuitive, spend a few minutes becoming familiar with the various options and buttons. 

**Question/task 8:** Identify the Haiti outbreak genomes in the tree. 

<div class="toggle"><button>Hint</button>
They are **most** of the non-black tips.
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - example tree is here
Question - Experiment with the different tree visualisation setting, which do you prefer? The different tree settings look like this:

**Question 9:** Are the isolates from Haiti monophyletic?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - yes.
Microreact is an ideal way to investigate relationships between phylogeny and geography (aka phylogeographical relationships). 

** Question 10:** Is there phylogeographical signal in the Cholera genomes from Haiti? You will have to zoom in on Haiti to see this. 

<div class="toggle"><button>Hint</button>
* Haiti is in the Caribbean.
* Or you can think of this another way, looking at the data, if you were given another isolate with no geographical information, how confident would you be in your prediction of it’s geographic origin?
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - not very strong. There is not very much diversity within the outbreak, which will always make it difficult for there to be geographical signal. There is a sub-clade of the outbreak separated by 1 SNP from the others, most of the isolates from northern Haiti are in here, but there are also some isolates from southern Haiti.

**Question 11:** Where is the closest neighbour of the Haiti genomes from?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - Nepal

**Question 12:** Write a short paragraph, summarising your results for the public health officials.

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

# Part 2 {-}

Inject 2 - Using the feedback we gave them on the likely origin of the cholera, public health officials have traced the likely source of the cholera outbreak to the UN Stabilisation Mission in Haiti Camp. A contingent of Nepalese soldiers had recently arrived in camp, and cholera is endemic in Nepal. Sewage from the camp is discharged into the Artibonite River, where many people draw their drinking water from. This is an incredibly sensitive subject because, while identifying the true source is very important, if the source is mis-identified, it could lead to a major diplomatic incident. Therefore, you have decided to search the public databases for all sequences from Nepal and Bangladesh and re-run the analysis, including these genomes. This will provide extra context for the Haitian outbreak and increase our certainty of what is happening.

On your CLIMB instance:
* Navigate to `~/data/outbreak_practical/phylo/dataset2`. 

* Run `ls` in this directory. This directory contains the alignment of all the high quality variable sites found in the core genome a for 19 *V. cholera* genomes from Haiti and 103 other *V. cholera* genomes from around the world by `snippy-core`. In terms of the inclusion from other countries, we have focussed on Isolates from the Indian sub-continent.

* Run IQ-TREE to generate a Maximum Likelihood tree
`iqtree -s dataset1.aln -nt AUTO -bb 100` 

<div class="alert alert-info">
* Note: It takes around 5 minutes. 
* Hint: be like a real bioinformatician, use this time to check twitter
</div>

**Question 1:** What is different between this command and the last time we ran `iqtree`? What does this option mean?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - `-bb 1000` ultrafast bootstrapping with 1000 re-samples, means each node in the tree 
* Then we need to run a couple of commands to convert the tree to a good format to visualise (we will midpoint root the tree and order the nodes to be descending). For this we will use a custom python script which uses the `ete3` library and the very useful Newick Utilities package.
```{bash, eval=FALSE}
cat all_vchol.aln.treefile | python ~/scripts/midpoint_root_tree.py | nw_order -c n - > all_vchol.aln.pretty.tree
```

* Then download the `all_vchol.aln.pretty.tree` onto your local computer and open it with `FigTree`. 

* `In FigTree`, go to `File -> Import annotations` in the menu bar, and navigate to the `outbreak_practical/phylo/dataset2/annotation_figtree.txt` file. Select this file and press open. Then in the main `FigTree` window, go to `Tip Lables`, and in the drop down `Display` menu, select ANNOT. You should now be able to see tip labels containing information about continent, country and year on the tree.

**Question 2:** SRR308716 was the most closely related non-Haitian genome to the Haiti outbreak strains in dataset 1. Is it still the most closely related genome?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - No, there are some very closely related genomes from Nepal e.g. SRR308715


**Question 4:** Do all the genomes from Nepal form a monophyletic group? What does this tell you about cholera in Nepal?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - No. Tells you that there are multiple circulating cholera clones/clades in Nepal.

**Question 5:** What does the root of the tree represent?

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - the inferred ancestor of all the V. cholera in our tree (not the reference genome)


<div class="alert alert-info">
Add in question on assessing bootstrap support - TODO
</div>


**Question 6:**  What is the impact of this extra context on your report to the public health doctors? Write a short paragraph reporting your conclusions.

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - We have confirmed that the Haitian outbreak strain shares a recent common ancestor with South Asian/Nepalese genomes. We are now confident that this geographical region is the likely origin of this outbreak.

**Extra task:**  Use the `phandango` tool to visualise the meta-data in a more interesting way. Navigate to here. And drag and drop your tree and the `annotation_phandango.csv` file onto the web page. Pressing `k` will bring up the key. Further instructions are here.

**Advanced question 1:** Using the phylogenetic tree, estimate the SNP distance between SRR308691 and  SRR308715. A piece of paper might help you estimate the distance between the two tips. Remember you have to go from one tip, down to the MRCA, back to the other tip. Only ‘horizontal’ distance (if you are viewing the tree in rectangular format) counts. The other information you need to calculate this is the tree length, which is 0.029 (from IQTREE log) and the alignment length, which is 987 bp (calculate dby `bioawk -c fastx '{ print $name, length($seq) }' < dataset2.aln`).

<div class="toggle"><button>Hint</button>
Comes from this formula: (`distance between nodes` / `total tree length`) * `alignment length`
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - (0.0012 / 0.029)  * 987 = 41 SNPs

**Advanced question 2:** How does this compare with `disty`?

<div class="toggle"><button>Hint</button>
 `grep -A 1 SRR308691 dataset2.aln > tmp.fa; grep -A 1 SRR308715 dataset2.aln >> tmp.fa; disty tmp.fa`
</div><br>

<textarea id="name" name="name" cols="100" rows="3" placeholder="You can input your answer here..."/>

Answer - 34 SNPs. Why is it different? Evolutionary model, measurement error?

## Notes {-}
* Should mention about ‘what is a useful cluster’ being a subjective measure (though some clustering methods do exist)
https://docs.google.com/presentation/d/1QFpPlBkIRSd977WO1tNp0DQMwJJIKtga9cjhNRCIlL4/edit#slide=id.g22fa57d6c6_0_635
https://www.dropbox.com/s/6r3hv1yg277igz2/RSLee_Workshop_I_instructions_20180609_no_drop.pdf

* Data selection - random selected 46 non-{Nepal,Bangladesh,Haiti} from Donman et al.
https://mbio.asm.org/content/5/6/e01721-14

https://www.ncbi.nlm.nih.gov/pubmed/29123068

* Put in the talk about trees as mobiles, same tree as unsorted, ascending, descending, circle, rectangular, etc. 

* Put in about identifying ‘signal’ in phylogenies into talk, in the context of prediction (strong signal, easy to predict where metadata not known)

<script>
  $(".toggle").click(function() {
    $(this).toggleClass("open");
  });
</script>
